### PR TITLE
feat: Added BreadCrumbs in Docs Page (#7045)

### LINKED ISSUE #7045: [FEATURE] Back button in the docs
### Description

Breadcrumbs on the main docs page.
When we open a particular doc, there should be a breadcrumb on that page also that acts as a back button.

We have Tailwind components, I suggest using this one...

Screenshot 2023-05-21 at 15 26 46

```js
import { ChevronRightIcon, HomeIcon } from '@heroicons/react/20/solid'

const pages = [
  { name: 'Projects', href: '#', current: false },
  { name: 'Project Nero', href: '#', current: true },
]

export default function Example() {
  return (
    <nav className="flex" aria-label="Breadcrumb">
      <ol role="list" className="flex items-center space-x-4">
        <li>
          <div>
            <a href="#" className="text-gray-400 hover:text-gray-500">
              <HomeIcon className="h-5 w-5 flex-shrink-0" aria-hidden="true" />
              <span className="sr-only">Home</span>
            </a>
          </div>
        </li>
        {pages.map((page) => (
          <li key={page.name}>
            <div className="flex items-center">
              <ChevronRightIcon className="h-5 w-5 flex-shrink-0 text-gray-400" aria-hidden="true" />
              <a
                href={page.href}
                className="ml-4 text-sm font-medium text-gray-500 hover:text-gray-700"
                aria-current={page.current ? 'page' : undefined}
              >
                {page.name}
              </a>
            </div>
          </li>
        ))}
      </ol>
    </nav>
  )
}
```

https://tailwindui.com/components/application-ui/navigation/breadcrumbs#component-7dc6ffff20237868ddf0e7d21cf17cf1

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by nitheesh-daram**
Closes #7045 

## Changes proposed

1. Added BreadCrumb to Docs Page to Easily Go back.
2. Made it Sticky for UX purposes.(if not needed we can Remove it).
3. Added id tag to Sections block in root Docs page to get back to the respective section using breadcrumb.
4. Yet to Finalize the colour scheme (I'm not good with colours :-) ).

## Check List (Check all the applicable boxes) 

- [X] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [X] This PR does not contain plagiarized content.
- [X] The title of my pull request is a short description of the requested changes.

## Screenshots
Light Mode :
image
image
Dark Mode
image
image

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/7160"></a>

### CODE REVIEW

**Comment by eddiejaoude** [comment]
Thank you for the contribution 👍 

Some thoughts on the design...
- the example provided in the issue didn't have a background colour, I find this clashes with the main navbar
- I don't think it needs to stick to the top (maybe thats why the background was added?)
- The `package-lock.json` files are not needed

File: components/BreadCrumb.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import { ChevronRightIcon, HomeIcon } from '@heroicons/react/20/solid'
+
+export default function BreadCrumb({pageData}) {
+
+  return (
+    <nav className="flex sticky top-2 bg-slate-900 dark:bg-slate-800 p-3 rounded z-[100]" aria-label="Breadcrumb">
+      <ol role="list" className="flex items-center space-x-4">
+        <li>
+          <div>
+            <Link href="/docs" className="text-gray-400 hover:text-gray-500">
```
Comment (eddiejaoude) [inline components/BreadCrumb.js:11]: Please use the colours from the tailwind config file

File: pages/docs/index.js [hunk 2]
```diff
@@ -374,7 +374,7 @@ export default function DocsIndex() {
           >
             <div className="relative mx-auto max-w-lg divide-y-2 divide-primary-low dark:divide-primary-low-high lg:max-w-7xl">
               <div>
-                <h2 className="text-3xl font-bold tracking-tight text-primary-high dark:text-primary-low sm:text-4xl">
+                <h2 className="text-3xl font-bold tracking-tight text-primary-high dark:text-primary-low sm:text-4xl" id = {section.title}>
```
Comment (eddiejaoude) [inline pages/docs/index.js:377]:
```suggestion
                <h2 className="text-3xl font-bold tracking-tight text-primary-high dark:text-primary-low sm:text-4xl" id={section.title}>
```

File: components/BreadCrumb.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import Link from '@components/Link'
+import { ChevronRightIcon, HomeIcon } from '@heroicons/react/20/solid'
```
Comment (eddiejaoude) [inline components/BreadCrumb.js:2]: We use react icons, you will be able to find the same icons there https://react-icons.github.io/react-icons/search?q=chevronright

**Comment by nitheesh-daram** [comment]
> Thank you for the contribution +1
> 
> Some thoughts on the design...
> 
> * the example provided in the issue didn't have a background colour, I find this clashes with the main navbar
> * I don't think it needs to stick to the top (maybe thats why the background was added?)
> * The `package-lock.json` files are not needed

Hey Eddie,Changes Done as per  Code Review.

Light
image

Dark:
image

File: components/BreadCrumb.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import Link from '@components/Link'
+import { FiChevronRight } from 'react-icons/fi'
+import {AiOutlineHome} from 'react-icons/ai'
+
+export default function BreadCrumb({pageData}) {
+
+  return (
+    <nav className="flex  p-3 " aria-label="Breadcrumb">
```
Comment (eddiejaoude) [inline components/BreadCrumb.js:8]: I am not sure if we need `nav` here, or if we only use it for the main navigation 🤔 - I will ask in EddieHub Discord

Btw the spacing doesn't look quite right
```suggestion
    <nav className="flex p-3" aria-label="Breadcrumb">
```

File: pages/docs/how-to-guides/editing.mdx [hunk 5]
```diff
@@ -103,7 +103,7 @@ To make further changes to an open Pull Request, follow these steps:
 GitHub will automatically append these changes to your open Pull Request.
 
 export default ({ children }) => (
-  <DocsLayout title="LinkFree Editing Profile Documentation">
+  <DocsLayout title="LinkFree Editing Profile Documentation" section="Customising your Profile" name="Editing Profile">
```
Comment (SaraJaoude) [inline pages/docs/how-to-guides/editing.mdx:106]:
```suggestion
  <DocsLayout title="LinkFree Editing Profile Documentation" section="Customising your Profile" name="Editing">
```

File: pages/docs/advanced/single-user-mode.mdx [hunk 6]
```diff
@@ -19,7 +19,7 @@ This allows you to host LinkFree yourself for your single Profile, but this will
 3. Host the app on your favourite platform
 
 export default ({ children }) => (
-  <DocsLayout title="LinkFree Full Example documentation">
+  <DocsLayout title="LinkFree Full Example documentation" section="Extra" name="single-user-mode">
```
Comment (SaraJaoude) [inline pages/docs/advanced/single-user-mode.mdx:22]:
```suggestion
  <DocsLayout title="LinkFree Full Example documentation" section="Extra" name="Single User Mode">
```

File: pages/docs/map.mdx [hunk 7]
```diff
@@ -33,5 +33,5 @@ You can find the relevant code that retrieves the location information for each
 Note: This situation only affects the local development environment and not the production environment. In production, the user markers will appear on the map since the location information of all users is already available in the database.
 
 export default ({ children }) => (
-  <DocsLayout title="LinkFree Map documentation">{children}</DocsLayout>
+  <DocsLayout title="LinkFree Map documentation" section="Extra" name="Map">{children}</DocsLayout>
```
Comment (SaraJaoude) [inline pages/docs/map.mdx:36]:
```suggestion
  <DocsLayout title="LinkFree Map documentation" section="Extra" name="Map Information">{children}</DocsLayout>
```

File: pages/docs/faqs.mdx [hunk 8]
```diff
@@ -24,5 +24,7 @@ Once your Pull Request is merged it is deployed automatically. The deployment ca
 If your Profile is not visible after merging, please contact the maintainers in <Link href="http://discord.eddiehub.org">EddieHub Discord</Link> or in the <Link href="http://github.com/EddieHubCommunity/LinkFree/discussions">GitHub Discussions in the LinkFree</Link> repo.
 
 export default ({ children }) => (
-  <DocsLayout title="LinkFree FAQ documentation">{children}</DocsLayout>
+  <DocsLayout title="LinkFree FAQ documentation" section="Extra" name="Faq">
```
Comment (SaraJaoude) [inline pages/docs/faqs.mdx:27]:
```suggestion
  <DocsLayout title="LinkFree FAQ documentation" section="Extra" name="FAQ">
```

File: pages/docs/contributing/automated-tests.mdx [hunk 9]
```diff
@@ -33,7 +33,7 @@ To open last HTML report run:
 ```
 
 export default ({ children }) => (
-  <DocsLayout title="LinkFree Automated Tests Documentation">
+  <DocsLayout title="LinkFree Automated Tests Documentation" section="Contributing" name="Automated Tests" >
```
Comment (eddiejaoude) [inline pages/docs/contributing/automated-tests.mdx:36]:
```suggestion
  <DocsLayout title="LinkFree Automated Tests Documentation" section="Contributing" name="Automated Tests">
```

File: pages/docs/contributing/reviewers.mdx [hunk 10]
```diff
@@ -99,5 +99,5 @@ The exception is if the changes have made modifications to the `package.json` fi
 If the author needs help undoing their changes to the lock file, I have created a video on how they can achieve this https://www.youtube.com/watch?v=jHkIXYVpMTo
 
 export default ({ children }) => (
-  <DocsLayout title="LinkFree Maintainers Documentation">{children}</DocsLayout>
+  <DocsLayout title="LinkFree Maintainers Documentation" section="Contributing" name="Reviewers" >{children}</DocsLayout>
```
Comment (eddiejaoude) [inline pages/docs/contributing/reviewers.mdx:102]:
```suggestion
  <DocsLayout title="LinkFree Maintainers Documentation" section="Contributing" name="Reviewers">{children}</DocsLayout>
```

File: pages/docs/contributing/storybook.mdx [hunk 11]
```diff
@@ -22,5 +22,5 @@ _note: not all components have been added, this is a great way to contribute to
 1. Run `npm run storybook` to see the added components and visit `http://localhost:6006` to interact with them in the browser
 
 export default ({ children }) => (
-  <DocsLayout title="LinkFree StoryBook Documentation">{children}</DocsLayout>
+  <DocsLayout title="LinkFree StoryBook Documentation" section="Contributing" name="Storybook" >{children}</DocsLayout>
```
Comment (eddiejaoude) [inline pages/docs/contributing/storybook.mdx:25]:
```suggestion
  <DocsLayout title="LinkFree StoryBook Documentation" section="Contributing" name="Storybook">{children}</DocsLayout>
```

File: pages/docs/environments/linkfree-cli.mdx [hunk 12]
```diff
@@ -42,5 +42,5 @@ npm run cli
 If you find any kind of bug you can use the 5th option `🐛 Report a bug` in the main prompt to report the bug. You can also create an issue in the [LinkFree CLI](https://github.com/Pradumnasaraf/LinkFree-CLI/issues/new?assignees=&labels=bug&template=bug.yaml) repository.
 
 export default ({ children }) => (
-  <DocsLayout title="Creating profile via LinkFree CLI">{children}</DocsLayout>
+  <DocsLayout title="Creating profile via LinkFree CLI" section="Getting started" name="LinkFree CLI" >{children}</DocsLayout>
```
Comment (eddiejaoude) [inline pages/docs/environments/linkfree-cli.mdx:45]:
```suggestion
  <DocsLayout title="Creating profile via LinkFree CLI" section="Getting started" name="LinkFree CLI">{children}</DocsLayout>
```

File: pages/docs/how-to-guides/milestones.mdx [hunk 13]
```diff
@@ -59,5 +59,5 @@ This is what a complete example looks like:
 3. Now you can commit your file and create a Pull Request, for more details please see <Link href="/docs/how-to-guides/editing">Editing Guide</Link>
 
 export default ({ children }) => (
-  <DocsLayout title="LinkFree Milestones Documentation">{children}</DocsLayout>
+  <DocsLayout title="LinkFree Milestones Documentation" section="Customising your Profile" name="Milestones" >{children}</DocsLayout>
```
Comment (eddiejaoude) [inline pages/docs/how-to-guides/milestones.mdx:62]:
```suggestion
  <DocsLayout title="LinkFree Milestones Documentation" section="Customising your Profile" name="Milestones">{children}</DocsLayout>
```

File: components/BreadCrumb.js [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+                className="ml-4 text-sm font-medium text-primary hover:text-primary-medium-low"
+              >
+                {pageData.name}
+              </Link>
+            </div>
+          </li>
+      </ol>
+    </nav>
+  )
+}
```
Comment (eddiejaoude) [inline components/BreadCrumb.js:43]: I think the formatting is not aligned to our project style, but I can fix in a temporary PR

**Review by eddiejaoude — APPROVED** [review]
Looks good 👍  thank you!

note: please check your Prettier plugin in VScode, your files are not formatted correctly, but I will fix via a temporary branch

**Comment by eddiejaoude** [comment]
Changes made in PR https://github.com/EddieHubCommunity/LinkFree/pull/7207

**Pull request merged**